data/transactions/stateproof.go

```diff
@@ -18,50 +18,56 @@ package transactions
 
 import (
 	"github.com/algorand/go-algorand/crypto"
-	"github.com/algorand/go-algorand/crypto/compactcert"
+	"github.com/algorand/go-algorand/crypto/stateproof"
 	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/stateproofmsg"
 	"github.com/algorand/go-algorand/protocol"
 )
 
-// CompactCertTxnFields captures the fields used for compact cert transactions.
-type CompactCertTxnFields struct {
+// StateProofTxnFields captures the fields used for stateproof transactions.
+type StateProofTxnFields struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	CertRound basics.Round             `codec:"certrnd"`
-	CertType  protocol.CompactCertType `codec:"certtype"`
-	Cert      compactcert.Cert         `codec:"cert"`
+	StateProofIntervalLatestRound basics.Round            `codec:"sprnd"`
+	StateProofType                protocol.StateProofType `codec:"sptype"`
+	StateProof                    stateproof.StateProof   `codec:"sp"`
+	Message                       stateproofmsg.Message   `codec:"spmsg"`
 }
 
-// Empty returns whether the CompactCertTxnFields are all zero,
+// Empty returns whether the StateProofTxnFields are all zero,
 // in the sense of being omitted in a msgpack encoding.
-func (cc CompactCertTxnFields) Empty() bool {
-	if cc.CertRound != 0 {
+func (sp StateProofTxnFields) Empty() bool {
+	if sp.StateProofIntervalLatestRound != 0 {
 		return false
 	}
-	if !cc.Cert.SigCommit.IsEmpty() || cc.Cert.SignedWeight != 0 {
+	if !sp.StateProof.SigCommit.IsEmpty() || sp.StateProof.SignedWeight != 0 {
 		return false
 	}
-	if len(cc.Cert.SigProofs.Path) != 0 || len(cc.Cert.PartProofs.Path) != 0 {
+	if len(sp.StateProof.SigProofs.Path) != 0 || len(sp.StateProof.PartProofs.Path) != 0 {
 		return false
 	}
-	if len(cc.Cert.Reveals) != 0 {
+	if len(sp.StateProof.Reveals) != 0 {
```

Comment:
We have a problem here. We've added `PositionsToReveal` to the state proof and here we don't make sure that this is zero.
I suggest it would be safer to call `sp.StateProof.MsgIsZero`

Reply:
This function is way too verbose either way, I'll refactor it